src/lib.rs

```diff
@@ -96,20 +97,146 @@ pub struct SelectionOutput {
 pub fn select_coin_bnb(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
+    rng: &mut ThreadRng,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut selected_inputs: Vec<usize> = vec![];
+    let bnb_tries = 1000000;
+
+    let mut sorted_inputs: Vec<(usize, OutputGroup)> = inputs
+        .iter()
+        .enumerate()
+        .map(|(index, input)| (index, input.clone()))
+        .collect();
+    sorted_inputs.sort_by_key(|(_, input)| std::cmp::Reverse(input.value));
+
+    let bnb_selected_coin = bnb(
+        &sorted_inputs,
+        &mut selected_inputs,
+        0,
+        0,
+        bnb_tries,
+        &options,
+        rng,
+    );
+    match bnb_selected_coin {
+        Some(selected_coin) => {
+            let accumulated_value: u64 = selected_coin
+                .iter()
+                .fold(0, |acc, &i| acc + inputs[i].value);
+            let accumulated_weight: u32 = selected_coin
+                .iter()
+                .fold(0, |acc, &i| acc + inputs[i].weight);
+            let estimated_fee = 0;
+            let waste = calculate_waste(
+                inputs,
+                &selected_inputs,
+                &options,
+                accumulated_value,
+                accumulated_weight,
+                estimated_fee,
+            );
+            let selection_output = SelectionOutput {
+                selected_inputs: selected_coin,
+                waste: WasteMetric(waste),
+            };
+            Ok(selection_output)
+        }
+        None => select_coin_srd(inputs, options, &mut rand::thread_rng()),
```

Comment:
Should be Err(NoSolutionFound)

Reply:
The reasoning behind this is when the user calls the high level API, srd will be invoked regardless of the result of bnb.
So when a user calls bnb in particular signalling no solution found is better than returning the result of srd